diff --git a/CMakeLists.txt b/CMakeLists.txt
index ec1cfa782..bb95041d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -386,6 +386,9 @@ if(CMAKE_USE_OPENSSL)
   set(SSL_ENABLED ON)
   set(USE_OPENSSL ON)
 
+  list(APPEND OPENSSL_LIBRARIES rt)
+  list(APPEND CURL_LIBS rt)
+
   # Depend on OpenSSL via imported targets if supported by the running
   # version of CMake.  This allows our dependents to get our dependencies
   # transitively.
