diff --git a/posix-configs/SITL/init/ekf2/iris b/posix-configs/SITL/init/ekf2/iris
index 91ebb0b3b8..d5d7dad3ca 100644
--- a/posix-configs/SITL/init/ekf2/iris
+++ b/posix-configs/SITL/init/ekf2/iris
@@ -61,8 +61,8 @@ ekf2 start
 mc_pos_control start
 mc_att_control start
 mixer load /dev/pwm_output0 ROMFS/px4fmu_common/mixers/quad_w.main.mix
-mavlink start -x -u 14556 -r 4000000
-mavlink start -x -u 14557 -r 4000000 -m onboard -o 14540
+mavlink start -x -u 14556 -r 4000000 -t 172.17.0.1
+mavlink start -x -u 14557 -r 4000000 -m onboard -o 14540 -t 172.17.0.1
 mavlink stream -r 50 -s POSITION_TARGET_LOCAL_NED -u 14556
 mavlink stream -r 50 -s LOCAL_POSITION_NED -u 14556
 mavlink stream -r 50 -s GLOBAL_POSITION_INT -u 14556
diff --git a/src/modules/uavcan/libuavcan b/src/modules/uavcan/libuavcan
--- a/src/modules/uavcan/libuavcan
+++ b/src/modules/uavcan/libuavcan
@@ -1 +1 @@
-Subproject commit 231b221b64265db4e98f86e53721316f9e17e143
+Subproject commit 231b221b64265db4e98f86e53721316f9e17e143-dirty
