diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0bb61919..534509ac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,17 +18,6 @@ endif()
 
 find_package(Python COMPONENTS Interpreter REQUIRED)
 
-# We automatically install the pip dependencies locally below.
-# Therefore, we check whether pip is available here.
-execute_process(
-    COMMAND ${Python_EXECUTABLE} -m pip -V
-    RESULT_VARIABLE EXIT_CODE
-    OUTPUT_QUIET
-)
-if (NOT ${EXIT_CODE} EQUAL 0)
-    message(FATAL_ERROR "Python pip not found, pip is required")
-endif()
-
 if (NOT MAVLINK_DIALECT)
     set(MAVLINK_DIALECT common)
 endif()
@@ -43,7 +32,6 @@ set(EXAMPLE_HEADER ${CMAKE_CURRENT_BINARY_DIR}/include/mavlink/${MAVLINK_DIALECT
 
 add_custom_command(OUTPUT ${EXAMPLE_HEADER}
     COMMAND ${Python_EXECUTABLE}
-        -m pip install -r pymavlink/requirements.txt --upgrade -t ${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/
         COMMAND ${CMAKE_COMMAND} -E env "PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}/pip-dependencies/" ${Python_EXECUTABLE}
         -m pymavlink.tools.mavgen
         --lang=C
